### PR TITLE
lint: ensure that TestCrlfmt errors are not split

### LINKED ISSUE #26928: testutils/lint: crlfmt's diff output is broken
Since 40c2b39 lint errors have extra newlines in them. This doesn't work well for crlfmt's output, which is supposed to be a diff. The extra newlines make it very hard to parse.

```
    --- FAIL: TestLint/TestCrlfmt (4.24s)
    	lint_test.go:728: 
    		/Users/benesch/go/src/github.com/cockroachdb/cockroach/pkg/storage/batcheval/cmd_end_transaction.go:1035
    	lint_test.go:728: 
    		-func changeReplicasTrigger(
    	lint_test.go:728: 
    		-	ctx context.Context,
    	lint_test.go:728: 
    		-	rec EvalContext,
    	lint_test.go:728: 
    		-	batch engine.Batch,
    	lint_test.go:728: 
    		-	change *roachpb.ChangeReplicasTrigger,
    	lint_test.go:728: 
    		-) result.Result {
    	lint_test.go:728: 
    		+func changeReplicasTrigger(
    	lint_test.go:728: 
    		+	ctx context.Context, rec EvalContext, batch engine.Batch, change *roachpb.ChangeReplicasTrigger,
    	lint_test.go:728: 
    		+) result.Result {
    	lint_test.go:728: 
    	lint_test.go:728: 
```

### PR DESCRIPTION
**Pull request by knz**
Fixes  #26928

There's an unrelated problem, that `crlfmt` always prints out color escape sequences even when its output is not a terminal, but we'll fix that separately.

### CODE REVIEW

**Review by benesch — APPROVED** [review]
Whoa, thanks for the quick turnaround!

**Comment by knz** [comment]
You're welcome!



bors r+



-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
